how to create and edit ACCESS-rOM3-panan documentation

chrisb13 opened the pull request:
## Preview link
[HERE](https://access-om3-configs.access-hive.org.au/pr-previews/573/configurations/pan-Antartic/Overview/)

-------------------

As part of this [discussion](https://github.com/ACCESS-NRI/access-om3-configs/issues/561), this is the start of making a fully supported ACCESS-rOM3-panan documentation. ping @claireyung @helenmacdonald.

Here's some steps to edit this on a terminal:
```bash
git clone git@github.com:ACCESS-NRI/access-om3-configs.git
cd access-om3-configs
git branch -vva
git checkout remotes/origin/cbull/561-ice-shelf-maindocs
git switch -c cbull/561-ice-shelf-maindocs
#edit the new /access-om3-configs/documentation/docs/pages/IceShelves.md
#make changes
git add IceShelves.md
git commit -m 'message'
git push --set-upstream origin cbull/561-ice-shelf-maindocs
```
A PR-preview will then appear in this draft PR. 

Alternatively just edit `IceShelves.md` online here and use GitHub to commit.

For community contributions they should make a fork and then merge back into this branch (I think!).

@claireyung, I think for now feel free to put all Ice Shelf documentation in here, we can wait till @atteggiani gets back to include configuration specific information in a "stub".

NB: not looking for a review just yet!

edoyango committed:
add IO optimization info

edoyango committed:
add compiler flag info

edoyango committed:
add runseq and cice notes

edoyango committed:
add a bit more detail

minghangli-uni commented:
Thanks @edoyango - looks like there’s an issue with the rendering:

<img width="1300" height="460" alt="image" src="https://github.com/user-attachments/assets/ee544f8b-c4dc-48e3-99ea-eacd3f865670" />


Also, feel free to rebase onto the latest commit when you’re ready.

chrisb13 commented:
> Hi @chrisb13, @minghangli-uni, @claireyung, @manodeep, I've added some notes on optimization to the iceshelf page. Please comment or add stuff if I've missed anything, gotten anything wrong, or something needs clarifying!

Amazing, thanks @edoyango! Re: [this](https://github.com/ACCESS-NRI/access-om3-configs/pull/573#issuecomment-3116336952) -- Seems to be rendering okay now after your recent commits.

chrisb13 committed:
test

claireyung committed:
Add some docs on panant no ice shelf setup

claireyung committed:
Make the links clickable

claireyung committed:
Fix some links and add sea ice categories

chrisb13 commented:
@claireyung so awesome to see this coming together!

On a small piece of interim feedback, can I suggest you also include the very detailed instructions in this doc' as well? They can just be included verbatim at this point.

I'm referring to this line:
> Detailed step-by-step instructions are provided [here](https://github.com/claireyung/access-om3-configs/blob/8km_jra_ryf_obc2-sapphirerapid-Charrassin-newparams-rerun-Wright-spinup-accessom2IC-yr9/panantarctic_instructions.md)

I mentioned this to @helenmacdonald and [having these kinds of detail level instructions](https://github.com/claireyung/access-om3-configs/blob/8km_jra_ryf_obc2-sapphirerapid-Charrassin-newparams-rerun-Wright-spinup-accessom2IC-yr9/panantarctic_instructions.md) is super helpful for other regional users interested in using om3 output, so I think it's really valuable to include here. (Some of this content will eventually be moved to a "stub" so that it only refers to this configuration but that's something we'll implement a bit later.)

claireyung committed:
Add new page with detailed instructions

claireyung commented:
@chrisb13 I directly copied my notes into a new subpage. Would you please be able to do the merging again as I think I broke the rendering now

claireyung commented:
Woops sorry I forgot to add the edits to the commit. Hopefully it's correct syntax but I can't really tell without the preview :)

atteggiani commented:
> I think it [might be related to having an open PR over here](https://github.com/ACCESS-NRI/access-om3-configs/pull/706) but I'll let @atteggiani correct me...

Correct!

atteggiani commented:
I just replied to those comments.

chrisb13 committed:
Update Detailedpananinstructions.md

Trivial change to force the pr-preview to try again.

chrisb13 commented:
@claireyung, we just merged, so seems to be working now:
 - https://access-om3-configs.access-hive.org.au/pr-preview-573/pages/IceShelves/
 - https://access-om3-configs.access-hive.org.au/pr-preview-573/pages/Detailedpananinstructions/
Had to make a silly commit ([2ca89fb](https://github.com/ACCESS-NRI/access-om3-configs/pull/573/commits/2ca89fbef845850e74c13342a9f3d8e8916d6890)) to make it build again. Look ok?

chrisb13 commented on documentation/docs/pages/Detailedpananinstructions.md:
```diff
@@ -0,0 +1,544 @@
+
+### OBC instructions
+
+Firstly, generate file using script. Needs hh5 analysis-22.10 at the moment. https://github.com/claireyung/mom6-panAn-iceshelf-tools/blob/main/generate-obcs/ACCESS-OM2_panan_boundary_forcing_005.ipynb
```
This link is broken. Perhaps this was the file that was meant:
https://github.com/claireyung/mom6-panAn-iceshelf-tools/blob/main/generate-obcs/ACCESS-OM2_panan_boundary_forcing_8km.ipynb
@claireyung ?

(Just out of curiosity, is `generate-obcs/ACCESS-OM2_panan_boundary_forcing_25km.ipynb` the file that was used for the 25km equivalent config? And perhaps this `generate-obcs/BYO-domain.ipynb` is from the regional mom6 workflow?)

chrisb13 commented:
@anton-seaice I'm aware that you're cleaning up the docs at the moment, perhaps we should aim to merge this soon? Maybe the same time that we approve the panan pr? Conveniently, it's not a stub but maybe should be in the configuration section? 

ping'ing @helenmacdonald as I think she has an interest in the how to create a regional config' bits...?

anton-seaice commented:
Yeah IceShelves.md is appropriate for Configurations section, and the instructions could maybe go in the Contributing section. Should not be impacted by any refactoring to remove stubs or use read the docs

chrisb13 commented:
> Yeah IceShelves.md is appropriate for Configurations section, and the instructions could maybe go in the Contributing section. Should not be impacted by any refactoring to remove stubs or use read the docs

Yeah I think that makes sense. Particularly whilst there isn't a similar regional repo'.

@helenmacdonald perhaps you'd be able to review the instructions / move it to Contributions?

chrisb13 commented:
> * @helenmacdonald can provide a review of the contents of `documentation/docs/pages/Detailedpananinstructions.md` in terms of helping people making their own regional configs. She'll likely have time next week.

@helenmacdonald just checking in on this, note [Anton's suggestions](https://github.com/ACCESS-NRI/access-om3-configs/pull/573#issuecomment-3515004433) -- I suggest we re-structure on that basis. Let me know if you need anything.

claireyung committed:
Update documentation/docs/pages/Detailedpananinstructions.md

Co-authored-by: Christopher Bull <5499680+chrisb13@users.noreply.github.com>

chrisb13 commented:
@helenmacdonald any thoughts on when you'll have a chance to look at this? Happy to help!

helenmacdonald committed:
Update IceShelves.md with clearer setup instructions

Clarified instructions for setting up a pan-Antarctic configuration, including details on model cloning, boundary conditions, and configuration file modifications.

helenmacdonald committed:
Revise Detailed Pan-An Instructions for Clarity

Updated instructions for creating and configuring 25km pan-An regional config. Improved clarity and corrected terminology throughout.

Still need to revise later sections

helenmacdonald committed:
Revise 8km domain setup instructions

Updated instructions for moving from a 25km to an 8km domain, including file copying, truncation, and configuration changes in various scripts and YAML files. Added notes on optimization and adjustments for new parameters.

helenmacdonald committed:
Fix typos and enhance clarity in documentation

Corrected typos and improved clarity in instructions regarding Charrassin bathymetry and model parameters. Added details on timestep restrictions and optimization improvements.

helenmacdonald committed:
Revise Detailedpananinstructions.md with introduction

Added introduction section and improved clarity of instructions.

helenmacdonald committed:
Revise pan-Antarctic instructions and update sections

Updated the structure and content of the pan-Antarctic regional domain instructions. Added new sections for optimization and parameter adjustments, and corrected the numbering of the steps.

chrisb13 commented:
@atteggiani, @helenmacdonald's last 3 commits seem to be failing the link checker, e.g.

```bash
Run lycheeverse/lychee-action@a8c4c7cb88f0c7386610c35eb25108e448569cb0
Run echo "$HOME/.local/bin" >> "$GITHUB_PATH"
Run # Remove any existing lychee binary to prevent conflicts
Run # Create a temporary directory for downloads and extraction
Downloading from: https://github.com/lycheeverse/lychee/releases/download/lychee-v0.21.0/lychee-x86_64-unknown-linux-gnu.tar.gz
Extracting lychee-x86_64-unknown-linux-gnu.tar.gz
lychee
docs/
docs/README.md
docs/PRE_COMMIT.md
docs/lychee.1
docs/TROUBLESHOOTING.md
Run # Install lychee from the temporary directory
Run /home/runner/work/_actions/lycheeverse/lychee-action/a8c4c7cb88f0c7386610c35eb25108e448569cb0/entrypoint.sh
Error: OR] Error while loading config: Cannot load configuration file `.github/workflows/lychee_config.toml`: No such file or directory (os error 2)

Error: Process completed with exit code 3.
```

any ideas?

chrisb13 committed:
part of the re-base, fixing some conflicts

chrisb13 commented:
@atteggiani so I think I did as you described but now it's [not rendering the page in the TOC](https://access-om3-configs.access-hive.org.au/pr-preview-573) despite [being listed there](https://github.com/ACCESS-NRI/access-om3-configs/pull/573/changes#diff-1426545775a936afe350c7f0417031796277de14abb8417a5640ad8b6d42a784). I also get a 404 on the landing page. Any ideas?

Perhaps I messed up resolving the many conflicts?

atteggiani commented:
> @atteggiani so I think I did as you described but now it's [not rendering the page in the TOC](https://access-om3-configs.access-hive.org.au/pr-preview-573) despite [being listed there](https://github.com/ACCESS-NRI/access-om3-configs/pull/573/changes#diff-1426545775a936afe350c7f0417031796277de14abb8417a5640ad8b6d42a784). I also get a 404 on the landing page. Any ideas?

Your "not rendering the page in the TOC" link points to https://access-om3-configs.access-hive.org.au/pr-preview-573 which is the old URL for PR previews. The preview now is at https://access-om3-configs.access-hive.org.au/pr-previews/573

I removed the comment with the old link.

helenmacdonald committed:
Moving IceShelves section to mkdocs configuration

Added IceShelves section with links to IceShelves and Detailed Ice shelf setup documentation.

helenmacdonald committed:
Simplify description for Detailed Ice shelf setup

helenmacdonald committed:
Refactor IceShelves section in mkdocs.yml

chrisb13 commented:
@claireyung. @helenmacdonald and I have just been going over the docs, you can see the [render here](https://access-om3-configs.access-hive.org.au/pr-previews/573/IceShelves/). 

From [this process](https://github.com/ACCESS-NRI/access-om3-configs/pull/814#issuecomment-3721380653), I've noticed that there's this [c0d8171](https://github.com/ACCESS-NRI/access-om3-configs/pull/814/commits/c0d81715deba6cd158b3a7917017926eab314d14) commit which looks to contain details of how you've been running the model? Should we also be including that in the config' doc? Very happy to do so but it's up to you. Relatedly, have there been any other instruction changes that we should be aware of? (Rather lucky, I spotted `c0d8171`.)

If you'd like to include it, can you please add the file to this branch (and remove it from `claireyung:dev-MC_4km_jra_ryf+regionalpanan+isf+draft`), this will keep your authorship.

chrisb13 committed:
addressing @claireyung's feedback https://github.com/ACCESS-NRI/access-om3-configs/pull/573#issuecomment-3721926994

chrisb13 commented:
> One comment - the configuration page name is `IceShelves` but neither of the files actually relate to the ice shelf config as they both were written for the pan-Antarctic om3 config without ice shelves. Should that be renamed to pan-Antarctic?

Great point! Now done [3182b77](https://github.com/ACCESS-NRI/access-om3-configs/pull/573/commits/3182b770a6dd7ef09645a48d4eea8079740c276b). [Rendered version](https://access-om3-configs.access-hive.org.au/pr-previews/573/configurations/pan-Antartic/Overview/).

> Related - I'm very happy to add the ice shelf how-to-run instructions in the docs (latest version [here](https://github.com/claireyung/access-om3-configs/blob/dev-MC_4km_jra_ryf%2Bregionalpanan%2Bisf%2Bdraft/ice_shelf_instructions.md) , but wondering where it should go given there is no other documentation yet on the ice shelf config? 

I've made a placeholder in this branch in commit [3182b77](https://github.com/ACCESS-NRI/access-om3-configs/pull/573/commits/3182b770a6dd7ef09645a48d4eea8079740c276b) (`access-om3-configs/documentation/mkdocs.yml`). Can you please add your file to this `Run ISF` location:

```yaml
        - pan-Antarctic:    
              - Overview: pages/configurations/pan-Antartic/Overview.md
              - Detailed: pages/configurations/pan-Antartic/Detailedpananinstructions.md
              - Run ISF: pages/configurations/pan-Antartic/run_panan_isf.md
```

(You can see that the `run isf` link [here](https://access-om3-configs.access-hive.org.au/pr-previews/573/configurations/pan-Antartic/Overview/) currently 404s)

> I started writing some for [mom6-sis2 ice shelves](https://github.com/claireyung/mom6-panAn-iceshelf-tools/wiki/How-to-make-a-new-panan-MOM6%E2%80%90SIS2-config-with-ice-shelves) a while ago, but haven't gotten around to it for the nuopc version yet, sorry!

This is a PR and I think we want to (eventually) have some documentation for both with / without ice shelves so I don't think it hurts to have it now.

@helenmacdonald has been reviewing this, so might want to chime in on what's best. Perhaps she's across what changes are needed for the NUOPC version? (We try to keep docs as short as possible but as long as necessary.)

claireyung committed:
Add how-to-run ice shelf docs

claireyung committed:
Add a copy of the ice shelf input files and config instructions, copied from https://github.com/claireyung/mom6-panAn-iceshelf-tools/wiki/ACCESS%E2%80%90OM3-config-with-ice-shelves-%E2%80%90-input-files-and-config-details

chrisb13 commented:
[751016f](https://github.com/ACCESS-NRI/access-om3-configs/pull/573/commits/751016f77ac9c2406aa7acccccb6dcf30608a19c) ❤️ !

claireyung commented:
Hi, i've updated my wiki with a nuopc version here https://github.com/claireyung/mom6-panAn-iceshelf-tools/wiki/ACCESS%E2%80%90OM3-config-with-ice-shelves-%E2%80%90-input-files-and-config-details and copied it into this branch in the above commit.

Some of the links in the page refer to pr previews of the panan no-ice shelf instructions so that probably needs to be fixed. I also added a bit of background overview of the config, and a few references.

Also, please note the instructions were sort of reverse-engineered because I didn't make the files in such a linear manner. They very likely are incomplete or have steps that don't work. Probably it would be good if someone tested the whole process on a new resolution so we can see what instructions don't work.... but that's also a big task so not sure if worth doing.

chrisb13 committed:
just adding @claireyung's latest doc update to the TOC so it is discoverable

chrisb13 commented:
@helenmacdonald / @claireyung, feel free to tweak the headings I came up with in this [169eabc](https://github.com/ACCESS-NRI/access-om3-configs/pull/573/commits/169eabcf9ccaf49240253dceb6fda52f280387b1).

As in:
[pan-Antarctic](https://access-om3-configs.access-hive.org.au/pr-previews/573/)
 - [Overview](https://access-om3-configs.access-hive.org.au/pr-previews/573/configurations/pan-Antartic/Overview/)
 - [Detailed noISF](https://access-om3-configs.access-hive.org.au/pr-previews/573/configurations/pan-Antartic/Detailedpananinstructions/)
 - [Detailed ISF](https://access-om3-configs.access-hive.org.au/pr-previews/573/configurations/pan-Antartic/Detailed_panan_isf/)
 - [Run ISF](https://access-om3-configs.access-hive.org.au/pr-previews/573/configurations/pan-Antartic/run_panan_isf/)

helenmacdonald committed:
Revise bathymetry and input file generation details

Updated the input file generation command and clarified initial conditions for the model.

helenmacdonald committed:
Update initial conditions and salt restoring references

helenmacdonald committed:
Revise boundary conditions details and add axis info

Updated boundary conditions section with notes on script compatibility and added axis information.

helenmacdonald committed:
Enhance documentation for NUOPC input and ice shelf config

Updated documentation for NUOPC input and ice shelf configuration setup, including links to relevant scripts and configuration details.
